<!-- xml -->
<launch>


  <arg name="map_name" default="test_map"/>
  <arg name="map_file" default="$(find medibotv4)/maps/$(arg map_name).yaml"/>
  <arg name="open_rviz" default="true"/>
  

  <!-- Run the map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" >
    <param name="frame_id" value="/map" /> 
  </node>

  <group ns="robot1">
    <arg name="robot_name"  value="robot1" />
    <arg name="tf_robot" value="$(arg robot_name)_tf" /> 
    <param name="tf_prefix" type="string" value="$(arg tf_robot)"/>

    <!--- AMCL -->
    <include file="$(find medibotv4)/launch/navigation/multirobot_amcl.launch">
      <arg name="robot_name" value="$(arg robot_name"/>
      <arg name="use_map_topic"             value="true"/>
      <arg name="global_frame_id" value="/map"/>
      <arg name="odom_frame_id"   value="$(arg tf_robot)/odom"/>
      <arg name="base_frame_id"   value="$(arg tf_robot)/base_link"/>
    </include>

    <!--move_base-->
    <include file="$(find medibotv4)/launch/navigation/multirobot_move_base.launch">
      <arg name="robot_name" value="$(arg robot_name)"/>
    </include>
 
  </group>
  
  <!-- <group ns="robot2">
    <arg name="robot_name"  value="robot2" />
    <arg name="tf_robot" value="$(arg robot_name)_tf" /> 
    <param name="tf_prefix" type="string" value="$(arg tf_robot)"/>
    
    <include file="$(find medibotv4)/launch/navigation/multirobot_move_base.launch">
     <arg name="robot_name" value="$(arg robot_name)"/>
     <arg name="tf_robot" value="$(arg tf_robot)" /> 
    </include>
    
  </group> -->

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find medibotv4_navigation)/rviz/multirobot_navigation.rviz"/>
  </group>
  
</launch>